### PR TITLE
prefs that require browser restart should show prompt for browser restart

### LINKED ISSUE #2543: prefs that require browser restart should show prompt for browser restart
There are a bunch of prefs in `about:preferences` that require browser restart. If they are changed, Brave should show a notification bar that says, 'Do you want to restart now?' and restart when 'Yes' is clicked.

### PR DESCRIPTION
**Pull request by darkdh**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

fix #2543

### CODE REVIEW

File: app/index.js [hunk 1]
```diff
@@ -381,6 +383,26 @@ app.on('ready', () => {
       app.quit()
     })
 
+    ipcMain.on(messages.PREFS_RESTART, () => {
+      var message = 'Do you want to restart now ?'
```
Comment (luixxiul) [inline app/index.js]: This should be translatable.

Comment (darkdh) [inline app/index.js]: Updated in 69bd164c2160a1ae2323d56b8cb8de40ebe930d6, thanks.

Comment (luixxiul) [inline app/index.js]: Thanks, and will you please include ``?`` too?

**Comment by darkdh** [comment]
Sure. Will do.

File: app/index.js [hunk 2]
```diff
@@ -381,6 +383,26 @@ app.on('ready', () => {
       app.quit()
     })
 
+    ipcMain.on(messages.PREFS_RESTART, () => {
+      var message = locale.translation('prefsRestart')
+
+      appActions.showMessageBox({
+        buttons: ['Yes', 'No'],
```
Comment (luixxiul) [inline app/index.js]: Should this also be translatable?

Comment (darkdh) [inline app/index.js]: Yes, and also update button of password prompt in 94d28c7

File: app/extensions/brave/locales/en-US/app.properties [hunk 3]
```diff
@@ -149,3 +149,7 @@ flashTitle=Flash Object Blocked
 flashRightClick=Right-click to run Adobe Flash
 flashSubtext=from {{source}} on {{site}}.
 flashExpirationText=Approvals reset 7 days after last visit.
+prefsRestart=Do you want to restart now ?
```
Comment (diracdeltas) [inline app/extensions/brave/locales/en-US/app.properties]: There's an extra space after 'now'

Comment (darkdh) [inline app/extensions/brave/locales/en-US/app.properties]: Thanks for reminding, @diracdeltas. Fixed!

**Comment by bbondy** [comment]
I think it would be nice to remove the Requires a restart text from the affected preferences as well because the text is too long and repetitive that way.  Maybe also say this: 

> This change requires a restart to come into effect.  Would you want to restart now?

**Comment by bbondy** [comment]
cc @bradleyrichter

**Comment by darkdh** [comment]
I came up with one scenario, what if the user made change to the pref required restart and ignore the notification bar and keep changing other prefs. Then this user may not know which pref required restart specifically.

**Comment by bbondy** [comment]
oh ok I see, we could do an electron MessageBox which is modal instead but we try to avoid those because it locks up everything completely.  Will defer to Brad.

File: js/about/preferences.js [hunk 4]
```diff
@@ -591,6 +592,10 @@ class AboutPreferences extends React.Component {
       settings: this.state.settings.set(key, value)
     })
     aboutActions.changeSetting(key, value)
+    if (key === settings.DO_NOT_TRACK || key === settings.HARDWARE_ACCELERATION_ENABLED ||
+      key === settings.PDFJS_ENABLED) {
```
Comment (diracdeltas) [inline js/about/preferences.js]: @bbondy was there a reason you added the restart requirement for PDFJS in e962ef99cb6ad70ec743f8911abb7be304a4b1bc? it seems to work fine when i turn it on/off and don't restart

Comment (bbondy) [inline js/about/preferences.js]: I'm not sure how to reproduce but sometimes even having the option on gives a save prompt. It's something related to flipping pref though. It works reliably with a restart without the bug.

**Comment by diracdeltas** [comment]
cherry-picked onto master, thanks

File: app/index.js [hunk 5]
```diff
@@ -381,6 +383,26 @@ app.on('ready', () => {
...
+
+      appActions.showMessageBox({
+        buttons: [locale.translation('yes'), locale.translation('no')],
+        options: {
+          persist: false
+        },
+        message
+      })
+      prefsRestartCallbacks[message] = (buttonIndex, persist) => {
+        if (buttonIndex === 0) {
```
Comment (diracdeltas) [inline app/index.js]: I think this is sometimes being called before the pref is actually saved. if i have flash enabled and disable in prefs then click 'yes' to restart, it is still enabled when Brave relaunches.

**Pull request closed without merging**